### PR TITLE
Added Windows Support

### LINKED ISSUE #3: Fix Security Alerts from GitHub against azuredevops-habitat repo
10 Open Security alerts of varying severity open against this repo - https://github.com/chef-partners/azuredevops-habitat/network/alerts, there's a need to lift a bunch of the dependency versions for this extension

### LINKED ISSUE #4: Add Windows support
Add support for running tasks on Windows build agents.

### LINKED ISSUE #5: Remove `vsts` from task names
VSTS is still used in the name of the task

This will be removed so it does not show up in build pipeline yaml files.

This will be a breaking change for anyone that is already using a YAML file for the build with the tasks. For this reason the extension will be bumped to version 3.

### PR DESCRIPTION
**Pull request by russellseymour**
A number of changes in this PR, the main change being that the extension now supports Windows.

# Windows Support

This has been enabled throughout the extension. It will **not** work on a hosted Windows agent because Habitat on Windows requires Administrator privileges.

Detection of the root privileges has been modified and resulted in the removal of a dependency.

Fixes #4 

# Exporting Habitat Build Variables

In previous tasks the full path to the `last_build.env` file was required. However the Windows equivalent is `last_build.ps1` which would not have been picked up. As the extension now detects the platform ht epath to the results directory is now required.

This is in the documentation and a warning is displayed in the logs if the file is detected in the path.

# Task Names

The string `vsts` has been removed from all the tasks. In fact there is now no dependency on the name of the pipeline.

Additionally this makes using the tasks in a pipeline yaml file easier. There is now new documentation on how to use the tasks in a YAML pipeline file.

Fixes #5 

Because of this change there has been a major version bump to version 3.

# Updated libraries

The libraries that the extension and tasks rely on have been updated. This fixes #3 

# Build badge

A build badge has been added to the repo and the overview file as seen in the Marketplace,

### CODE REVIEW

File: docs/pages/tasks/export-build-variables.md [hunk 1]
```diff
@@ -13,7 +13,9 @@ Exposes the last build variables to the Azure DevOps agent
 
 ## Description
 
-This task will read the contents of the `results/last_build.env` variable and expose the values as environment variables that can be used in subsequent tasks.
+Depending on the platform, this task will read the contents of the `results/last_build.env` or `results/last_build.ps1` variable and expose the values as environment variables that can be used in subsequent tasks.
```
Comment (stuartpreston) [inline docs/pages/tasks/export-build-variables.md]: Mention which one applies to which platform (i.e. add Linux and Windows)

File: package.json [hunk 2]
```diff
@@ -1,5 +1,5 @@
 {
-  "name": "chef-vsts-tasks",
+  "name": "habitat-tasks",
   "description": "Chef Integration for Visual Studio Team System (VSTS)",
```
Comment (stuartpreston) [inline package.json:3]: Update to Azure Pipelines

**Review by stuartpreston — APPROVED** [review]
Just a couple of naming comments really

File: docs/pages/tasks/export-build-variables.md (hunk 1 above)
Comment (russellseymour) [reply docs/pages/tasks/export-build-variables.md]: OK will do

File: package.json (hunk 2 above)
Comment (russellseymour) [reply package.json:3]: Damn missed one :-)

Comment (russellseymour) [reply package.json:3]: Sorry I didn't
I have taken the name of the build platform out so we are not dependent on it
Have a look at the docs for the pipeline and you will see why

**Review by stuartpreston — APPROVED** [review]

**Pull request merged**